deeplearning4j-core/src/main/java/org/deeplearning4j/nn/conf/layers/setup/ConvolutionLayerSetup.java

```diff
@@ -147,46 +147,44 @@ private void updateLayerInputs(int i, Layer inputLayer, Layer outputLayer){
                     }
                     break;
                 case "SubsamplingLayer":
-                    if(i < lastLayerNumber){
```

Comment:
Please keep this conditional because it ensures that subsampling is not the last layer.